processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -31,21 +32,34 @@ const (
 	TypeStr = "system"
 )
 
+var hostnameSourcesMap = map[string]func(*Detector) (string, error){
+	"dns": getFQDN,
+	"os":  getHostname,
+}
+
 var _ internal.Detector = (*Detector)(nil)
 
 // Detector is a system metadata detector
 type Detector struct {
-	provider systemMetadata
-	logger   *zap.Logger
+	provider        systemMetadata
+	logger          *zap.Logger
+	hostnameSources []string
 }
 
 // NewDetector creates a new system metadata detector
-func NewDetector(p component.ProcessorCreateSettings, _ internal.DetectorConfig) (internal.Detector, error) {
-	return &Detector{provider: &systemMetadataImpl{}, logger: p.Logger}, nil
+func NewDetector(p component.ProcessorCreateSettings, dcfg internal.DetectorConfig) (internal.Detector, error) {
+	cfg := dcfg.(Config)
+	if len(cfg.HostnameSources) == 0 {
+		cfg.HostnameSources = []string{"dns", "os"}
+	}
```

Comment:
Instead of patching the config here, can we add `system -> hostname_sources -> ["dns", "os"']` in `createDefaultConfig` even if `system` detector is not enabled by default?

Then we can also reject `hostname_sources: []` config during the validation step

Reply:
It will be done once https://github.com/open-telemetry/opentelemetry-collector/issues/4001 gets resolved. So, for now added `TODO` for the same.